crates/edr_provider/src/data.rs

```diff
@@ -555,10 +556,7 @@ impl ProviderData {
                 || {
                     let last_block = self.last_block()?;
 
-                    let base_fee = last_block
-                        .header()
-                        .base_fee_per_gas
-                        .unwrap_or_else(|| calculate_next_base_fee(last_block.header()));
+                    let base_fee = calculate_next_base_fee(last_block.header());
```

Comment:
This change was necessary to make the `should update the baseFeePerGas starting with the first block` and `should compute the next base fee correctly when ${hardfork} is activated` tests pass.

Reply:
Just making sure I understand correctly, but the original logic was incorrect and we should always run `calculate_next_base_fee` using the previous header - if `self.next_block_base_fee_per_gas` is not set?

Reply:
Yes, that's my understanding based on the expectations in the test and from reading: https://github.com/NomicFoundation/ethereumjs-vm/blob/60f4124daef005d913e6e7478c92d988f92d3a72/packages/block/src/header.ts#L452

We should only use the base fee from the previous block if the parent gas target equals the gas used in the parent block (`calculate_next_base_fee` handles this).

Reply:
Thanks!